src/Cache/__tests__/Cache-test.js

```diff
@@ -266,11 +268,10 @@ describe('Cache', () => {
         Promise.resolve('baz value')
       );
 
-      // jest has some trouble with promises and timeouts within promises :(
-      jest.runAllTimers();
-      jest.runAllTimers();
-
-      expect(fs.writeFile).toBeCalled();
+      setTimeout(() => {
```

Comment:
Not entirely sure what the root issue is, but I tried a variety of different solutions and it _only_ worked when I turned off the Jest timers. If you're looking to beef up the implementation, I highly recommend http://sinonjs.org/ for inspiration - their fake clock implementation is A++.